apps/webapp/app/routes/api.v1.runs.$runId.ts

```diff
@@ -0,0 +1,124 @@
+import type { LoaderArgs } from "@remix-run/server-runtime";
+import { json } from "@remix-run/server-runtime";
+import { cors } from "remix-utils";
+import { z } from "zod";
+import { prisma } from "~/db.server";
+import {
+  authenticateApiRequest,
+  getApiKeyFromRequest,
+} from "~/services/apiAuth.server";
+import { apiCors } from "~/utils/apiCors";
+import { taskListToTree } from "~/utils/taskListToTree";
+
+const ParamsSchema = z.object({
+  runId: z.string(),
+});
+
+const SearchQuerySchema = z.object({
+  cursor: z.string().optional(),
+  take: z.coerce.number().default(20),
+  subtasks: z.coerce.boolean().default(false),
+  taskdetails: z.coerce.boolean().default(false),
+});
+
+export async function loader({ request, params }: LoaderArgs) {
+  if (request.method.toUpperCase() === "OPTIONS") {
+    return apiCors(request, json({}));
+  }
+
+  const apiKeyResult = getApiKeyFromRequest(request);
+  const authenticatedEnv = await authenticateApiRequest(request, {
+    allowPublicKey: true,
+  });
+  if (!authenticatedEnv || !apiKeyResult) {
+    return apiCors(
+      request,
+      json({ error: "Invalid or Missing API key" }, { status: 401 })
+    );
+  }
+
+  const { runId } = ParamsSchema.parse(params);
+
+  const url = new URL(request.url);
+  const query = SearchQuerySchema.parse(Object.fromEntries(url.searchParams));
+  const showTaskDetails = query.taskdetails && apiKeyResult.type === "PRIVATE";
```

Comment:
Would be better if `authenticateApiRequest` returned the type of key instead of having to do the extra call the `getApiKeyFromRequest`